### PR TITLE
Feature upgrades: generator workflow, testing, and accessibility improvements

### PR DESCRIPTION
**Pull request by brianfunk**
## Summary
- implemented major generator workflow upgrades across data input, URL/session sharing, output validation, and Drupal package scaffolding
- added test automation with Vitest and wired tests into CI
- improved accessibility semantics (skip link, landmarks, aria states, live regions, heading order) and reduced homepage accessibility issues
- upgraded vulnerable dependencies and resolved audit findings

## Key Changes
- Generator UX/state: multi-mode data input (JSON/CSV/manual), share links, robust status/error handling
- Output system: download preflight validation + richer Drupal ZIP scaffold
- Performance: lazy-loaded heavy preview modules and improved chunking
- Quality: lint/test/build/audit gates passing

## Validation
- `npm test`
- `npm run lint`
- `npm run build`
- `npm audit --audit-level=high`

All commands pass locally on `dev`.

## Accessibility Notes
- generator route scans clean in axe for tested checks
- homepage still has remaining color-contrast findings tied to current visual palette; structural/semantic issues were addressed in this pass

### CODE REVIEW

File: src/pages/Generator.tsx [hunk 1]
```diff
@@ -42,6 +39,164 @@
...
+
+const decodeState = (value: string): unknown => {
+  return JSON.parse(decodeURIComponent(atob(value)));
+};
+
+const sanitizeForCodeGeneration = (obj: unknown): unknown => {
+  if (typeof obj === 'string') {
+    return obj
+      .replace(/<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi, '')
+      .replace(/javascript:/gi, '');
```
[the remark this answers is not in the document]
Comment (brianfunk) [reply src/pages/Generator.tsx]: Fixed in 7ad2bfc. Replaced the regex-based sanitization entirely with proper HTML entity encoding (`&`, `<`, `>`, `"`, `'`). This eliminates the incomplete URL scheme check since dangerous schemes like `javascript:`, `data:`, and `vbscript:` are neutralized by encoding their `:` and other characters.

File: src/pages/Generator.tsx [hunk 2]
```diff
@@ -42,6 +39,164 @@
...
+};
+
+const decodeState = (value: string): unknown => {
+  return JSON.parse(decodeURIComponent(atob(value)));
+};
+
+const sanitizeForCodeGeneration = (obj: unknown): unknown => {
+  if (typeof obj === 'string') {
+    return obj
+      .replace(/<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi, '')
```
[the remark this answers is not in the document]
Comment (brianfunk) [reply src/pages/Generator.tsx]: Fixed in 7ad2bfc. The regex-based script tag stripping has been replaced with HTML entity encoding, which is not susceptible to incomplete multi-character sanitization. All `<` and `>` characters are encoded to `&lt;` and `&gt;`, preventing any HTML element injection.

[the remark this answers is not in the document]
Comment (brianfunk) [reply src/pages/Generator.tsx]: Fixed in 7ad2bfc. Removed the problematic script tag regex entirely in favor of HTML entity encoding, which correctly handles all edge cases including `</script >` variants.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -30,6 +30,9 @@ jobs:
     - name: Run linting
       run: npm run lint
 
+    - name: Run unit tests
+      run: npm test
```
[the remark this answers is not in the document]
Comment (brianfunk) [reply .github/workflows/ci.yml:34]: Fixed in 7ad2bfc. Updated CI matrix from `[18.x, 20.x]` to `[20.x, 22.x]` since vitest@4.0.18 requires Node `^20.0.0 || ^22.0.0 || >=24.0.0`.

File: src/pages/Generator.tsx [hunk 4]
```diff
@@ -42,6 +39,164 @@ interface OutputFormat {
...
+
+    if (parsed && typeof parsed === 'object' && 'data' in parsed && Array.isArray((parsed as { data?: unknown }).data)) {
+      return normalizeMapRows((parsed as { data: TabularRow[] }).data);
+    }
+
+    throw new Error('Map JSON must be GeoJSON or an array with lat/lng values.');
+  }
+
+  if (isValidChartJsData(parsed)) {
+    return parsed as ChartJsLikeData;
```
[the remark this answers is not in the document]
Comment (brianfunk) [reply src/pages/Generator.tsx:188]: Fixed in 7ad2bfc. `isValidChartJsData` now treats `labels` as optional — datasets-only JSON with `{x,y}` points (scatter/bubble) is accepted. Labels/data length matching is only enforced when labels are present. Added test coverage for this case.
